client/components/MainSettings.jsx

```diff
@@ -19,45 +20,62 @@ const MainSettings = React.createClass({
     });
   },
 
-  componentWillUnmount() {
-    Meteor.call("room.archive", this.props.room._id, this.state.archived);
-  },
-
   toggleArchivedState() {
-    Meteor.call("room.archive", this.props.room._id, !this.state.archived);
+    const params = {
+      archived: !this.state.archived,
+      roomId: this.props.room._id
+    };
+    Meteor.call("room.archive", params);
     this.setState({archived: !this.state.archived});
```

Comment:
What if the meteor call fails? You will then have an inconsistency between server and client without knowing it. Can't you use the archived value of the room doc directly instead of having it in state? This applies in general as well.